File: Cpp/1464_Maximum_Product_of_Two_Elements_in_an_Array.cpp
class Solution {
public:
    int maxProduct(vector<int>& nums) {
        int first = max(nums[0], nums[1]);
        int second = min(nums[0], nums[1]);
        
        for(int i=2; i<nums.size(); ++i){
            if(nums[i]>=first){
                second = first;
                first = nums[i];
            } else if(nums[i]>second){
                second = nums[i];
            }
        }
        
        return (first-1)*(second-1);
    }
};

File: Kotlin/1464_Maximum_Product_of_Two_Elements_in_an_Array.kt
class Solution {
    fun maxProduct(nums: IntArray): Int {
        var first = Int.MIN_VALUE + 1
        var second = Int.MIN_VALUE
        
        nums.forEach{
            if(it >= first){
                second = first
                first = it
            } else if(it >= second){
                second = it
            }
        }
        
        return (first - 1) * (second - 1)
    }
}